js/models/messages.js

```diff
@@ -415,24 +415,57 @@
     },
 
     async acceptFriendRequest() {
+      const primaryDevicePubKey = this.attributes.conversationId;
+
       if (this.get('friendStatus') !== 'pending') {
         return;
       }
-      const conversation = await this.getSourceDeviceConversation();
-      // If we somehow received an old friend request (e.g. after having restored
-      // from seed, we won't be able to accept it, we should initiate our own
-      // friend request to reset the session:
-      if (conversation.get('sessionRestoreSeen')) {
-        conversation.sendMessage('', null, null, null, null, {
-          sessionRestoration: true,
-        });
-        return;
+
+      const allDevices = await libloki.storage.getAllDevicePubKeysForPrimaryPubKey(
+        primaryDevicePubKey
+      );
+
+      // Set profile name to primary conversation
+      let profileName;
+      const allConversationsWithUser = allDevices.map(d =>
+        ConversationController.get(d)
+      );
+      allConversationsWithUser.forEach(conversation => {
+        // If we somehow received an old friend request (e.g. after having restored
+        // from seed, we won't be able to accept it, we should initiate our own
+        // friend request to reset the session:
+        if (conversation.get('sessionRestoreSeen')) {
+          conversation.sendMessage('', null, null, null, null, {
+            sessionRestoration: true,
+          });
+          return;
+        }
+
+        profileName = conversation.getProfileName() || profileName;
+        conversation.onAcceptFriendRequest();
```

Comment:
This already loops through all devices associated with the contact, right? So invoking this for each of the contact's devices means you're going to send the friend request accepted message twice to each device 🤔

Reply:
Uh it doesn't.
```js
 async onAcceptFriendRequest(options = {}) {
      if (this.unlockTimer) {
        clearTimeout(this.unlockTimer);
      }
      if (this.hasReceivedFriendRequest()) {
        this.setFriendRequestStatus(FriendRequestStatusEnum.friends, options);

        await this.respondToAllFriendRequests({
          response: 'accepted',
          direction: 'incoming',
          status: ['pending', 'expired'],
        });
        window.libloki.api.sendBackgroundMessage(
          this.id,
          window.textsecure.OutgoingMessage.DebugMessageType
            .INCOMING_FR_ACCEPTED
        );
      }
    },
```

As seen from above it just sets its own friend request status and sends out a FR accept to the specific device.

Reply:
Ah, I was looking at conversations.js → `acceptFriendRequest()`